File: solutions/3430-count-days-without-meetings/solution.java
import java.util.Arrays;

class Solution {
    public int countDays(int days, int[][] m) {
        if (m.length == 0) return days;
        
        Arrays.sort(m, (a, b) -> a[0] - b[0]);
        int last = 0, res = 0;
        
        for (int[] meet : m) {
            if (meet[0] > last + 1) res += meet[0] - last - 1;
            last = Math.max(last, meet[1]);
            if (last >= days) return res;
        }
        
        return res + Math.max(days - last, 0);
    }
}

File: solutions/3430-count-days-without-meetings/solution.py
class Solution:
    def countDays(self, days: int, meetings: List[List[int]]) -> int:
        meetings.sort(key = itemgetter(0))

        i = 0
        s, e = meetings[0]
        while i + 1 < len(meetings):
            start, end = meetings[i]
            next_start, next_end = meetings[i + 1]
            merged: tuple[int, int]

            if end >= next_start:
                merged = start, max(end, next_end)
                meetings[i:i+2] = [merged]
            else:
                days -= end - start + 1
                i += 1
        else:
            last_start, last_end = meetings[-1]
            days -= last_end - last_start + 1
        return days
        
            
